internal/controller/volumereplicationgroup_controller.go

```diff
@@ -548,6 +556,10 @@ func (v *VRGInstance) processVRG() ctrl.Result {
 		return v.invalid(err, "Failed to process list of PVCs to protect", true)
 	}
 
+	if err := v.updatePVCListForCG(); err != nil {
```

Comment:
(future) At some point for clarity, we potentially should remove the optimizations for PVC grouping into either lists and let the separate handle it. As currently we look at various conditions and optimize out calling the separate function. This is to help code redability and not functionality as such.